EIPS/eip-2569.md

```diff
@@ -0,0 +1,357 @@
+
+One of the most significant reasons for not doing so is that in general the size of an image is much bigger than the size of a text file, thus the gas needed to save an image in Ethereum would exceed a block's gas limit. 
+
+However this changed a lot after the SVG(Scalable Vector Graphics) specification was developed by W3C since 1999. 
+
+The SVG specification offers several advantages (for more details about the advantages please refer to a reference link:https://en.wikipedia.org/wiki/Scalable_Vector_Graphics) over raster images. One of these advantages is its compact file-size.
+
+"Compact file-size – Pixel-based images are saved at a large size from the start because you can only retain the quality when you make the image smaller, but not when you make it larger. This can impact a site’s download speed. Since SVGs are scalable, they can be saved at a minimal file size".
+
+This feature well fixes the painpoint of saving an image file in Ethereum, therefore we think saving an SVG image in Ethereum is a good solution for keep the image permanent and tamper-resistant.
+
+In most ERC-721 related DAPPs they display an image for a non-fungible token. In most ERC-20 related DAPPs they don't have an image for a fungible token. We think displaying an image for a token either based on existing token standards such as ERC-20, ERC-721, ERC-1155 or based on future standards is needed in many use cases. Therefore those DAPPs which currently don't display an image for a token will eventually need such a function. 
+
+However with regard to most of the existing DAPPs which can display an image for a token they save such an image in a centralized server which, we think, is just a compromised solution. By utilizing the SVG specification we think converting a token's image to an SVG image and saving it in Ethereum provides a better solution for DAPPs to access an image for a token.
+
+This solution not only works for tokens based on ERC-721, ERC-1155 and ERC-20 but will work for tokens based on future standards. 
+
+## Backwards Compatibility
+There are no backward compatibility issues.
+
+## Test Cases
+
+## Implementation
+```tokenId```: a token index in an ERC-721 token or a token type/index in an ERC-1155 token. It is a uint256 variable.  
+
+```imageSvg```: an SVG image's file content. It is a string variable. Note: the SVG image should include at least three attributes:"name", "description" and "issuer".
+
+```setTokenImageSvg```: interface to set an SVG image to a token with or without an ID number.
+
+```getTokenImageSvg```: interface to get an SVG image for a token with or without an ID number.
+
+We propose to add three sol files in the existing ERC-721 implementation.
+Here are the details for the proposed sol files.
+
+```
```

Comment:
Solidity syntax highlighting will be added to the renderer soon, this will tell the renderer to render using Solidity syntax highlighting.
```suggestion
```solidity
```

Reply:
Changes made and committed